src/wp-admin/edit-form-advanced.php

```diff
@@ -491,6 +491,54 @@
 <?php
 if ( isset( $post_new_file ) && current_user_can( $post_type_object->cap->create_posts ) ) {
 	echo ' <a href="' . esc_url( admin_url( $post_new_file ) ) . '" class="page-title-action">' . esc_html( $post_type_object->labels->add_new ) . '</a>';
+	
+	/**
+	 * Adds Previous and Next buttons alongside Add New button.
+	 * 
+	 * @param bool 	Whether to show Previous and Next buttons. Default true.
+	 * @param $post_type
+	 *
+	 * @since CP-1.x.x
+	 */
+	if ( apply_filters( 'admin-post-navigation', true, $post_type_object->name ) ) {
+
+		// Setting per user
+		$nav_display = 'inline';
+		$metaboxhidden_post = get_user_meta( get_current_user_id(), 'metaboxhidden_post', true );
+
+		if ( in_array( 'adminpostnav', $metaboxhidden_post ) ) {
+			$nav_display = 'none';
+		}
+
+		/*
+		 * Enable overriding of default choice (made on post screen) for
+		 * other post types.
+		 * 
+		 * @param $nav_display	Must be either 'inline' or 'none'.
+		 * @param $post_type
+		 *
+		 * @since CP-1.x.x 
+		 */
+		$admin_nav = apply_filters( 'admin-post-navigation-buttons', $nav_display, $post_type_object->name );
+		$nav_display = ( in_array( $admin_nav, ['inline', 'none'] ) ) ? $admin_nav : $nav_display;
+
+		$previous_post = get_previous_post();
+		$next_post = get_next_post();
+
+		echo '<span id="adminpostnav" class="postbox" style="background-color: transparent; border: none; box-shadow: none; display: ' . $nav_display . ';">';
```

Comment:
The `display` property is inline because that's how the Screen Options work. I have kept as close as possible to what's already done so that it doesn't have any unintentional side-effects.

The other inline styles are there because using the `postbox` class necessary to hook into the correct code adds some unwanted visible effects. That's because this code is used for the metaboxes in the top line. I considered having this setting up there too, but it doesn't really fit because it's not a metabox.

Reply:
You can put the styles into an actual CSS file and simply manipulate the class.
If it's not a metabox, perhaps it shouldn't use the same code?

Reply:
> If it's not a metabox, perhaps it shouldn't use the same code?
That would mean writing a lot of entirely new code, because there isn't anything in the Screen Options that already behaves this way except for the metaboxes. I don't see any reason to do that when there's a perfectly viable system already in existence, with the only artefact it produces being a couple of visual effects that are easily overridden.

I don't see the problem with using inline styles here when the main functionality demands that they be used anyway.

Reply:
Ah, but the functionality is not dependent on the style.
Are you taking into account accessibility? Do these links have screen reader text? If using `display:none` is okay because you don't want the screen reader to see it either, then just define two classes with your styles with the two values for `display`, and put the appropriate class on it. That way, the styles are separate as they should be. If you think you *must* manipulate the style, at least use a custom property that you set in JS.

Reply:
> That way, the styles are separate as they should be.

Of course I could do it the way you've suggested, but who says they "should" be like that? As I said, inline styles are exactly how core handles this already and, yes, it's perfectly accessible. Nothing should be shown to a screen-reader if the buttons aren't displayed. And if you think that styles should always go in a separate class in a CSS file, why haven't you been consistent and said the same about the JavaScript?

Reply:
I have been working for awhile on admin color schemes and using custom properties, so of course when I see a color used in an inline style, I think that's a bad thing. Also, this could be updated to use flex or something and you have hard-coded the valid values into the script, whereas if it were a class that you manipulate, you can name it well and it would be self-documenting as well as easier to change (and for plugins to override).
As for the JS, yes I did wonder about it being inline, but I left it for someone else to comment on.

Reply:
I'm not using any colors at all in the inline-styles! The background color is set to transparent and the other two properties are border and box-shadow (both set to none), so they won't get in the way of what you're doing. This is actually a classic case of where inline styles make sense.

So too with the JS, because small snippets should normally be served inline. It also makes it much easier to find, and I've used vanilla JS rather than the awful jQuery that's in the `postbox.js` file (which uses `show` and `hide` though those functions are very much frowned on) .

Reply:
Last I checked, transparent was a color. :smile:
And inline styles are about impossible to override. It's a very poor example to follow, if that's the argument.

Reply:
It's not a color: it's clear! So you can see the color beneath — which is precisely the color you'd be interested in changing with your admin styles. So precisely no harm done.